src/coreclr/gc/gc.cpp

```diff
@@ -12674,7 +12748,11 @@ void gc_heap::distribute_free_regions()
             {
                 next_region = heap_segment_next (region);
                 int age_in_free_to_decommit = min (max (AGE_IN_FREE_TO_DECOMMIT, n_heaps), MAX_AGE_IN_FREE);
-                if (heap_segment_age_in_free (region) >= age_in_free_to_decommit)
+                if (
+                     (heap_segment_age_in_free (region) >= age_in_free_to_decommit)
+                     ||
+                     ((get_region_committed_size (region) == GC_PAGE_SIZE) && joined_last_gc_before_oom)
+                   )
```

Comment:
nit - styling. no need to put these on separate lines unless you have to check for #ifdef's...

```suggestion
                // when we are about to get OOM, we'd like to discount the free regions that just have the initial page commit as they are not useful
                if ((heap_segment_age_in_free (region) >= age_in_free_to_decommit) ||
                    ((get_region_committed_size (region) == GC_PAGE_SIZE) && joined_last_gc_before_oom))
```